### PR TITLE
Random showcase

### PR DESCRIPTION
**Pull request by seanyoung247**
# Hackathon Git Labs Project PR 🎉🎉

Adds a showcase slideshow as a web component. Ref: https://github.com/auxfuse/hackathon-git-labs/issues/150
Also refactors code for readability and reuse. Moves the helper functions from main.js to cards.js and utilities.js

I hope you guys like it. Personally, I'm pretty proud of this and it's going straight in my code library for future projects.

## PR contains:
- Adds files:
 - webcomponent.js - base class and helper functions for web components.
 - slide-show component:
   - slideshow.js - JavaScript class implementation
   - slideshow.css - component styles
   - slideshow.html - component shadow dom (internal) styles and html template
 - General:
   - cards.js - module of card handling functions
   - utilities.js - module of general utility functions
   - slideshowscroll.txt - iframe scrolling was removed, but this text file has the code that performs the action if desired.

slide-show component:
Defines custom element tag: `slide-show`. As a web-component with encapsulated shadow DOM this should be easily reusable just like a normal HTML tag.
- Conforms to progressive enhancement, if custom element fails the slide-show will display as a horizontally scroll-able list of slides, using CSS only.
- Allows theming and customization.
  - Element colors can be adjusted by targeting the `--fill-color` and `--stroke-color` CSS variables.
  - Some elements can be directly restyled using the ::part() selector: 
    - `slide-controls`, `previous-btn`, `next-btn`, `slide-indicator`, and `slide-indicator-list`
      - For instance: `slide-show::part(slide-controls) { display: none; }` will hide all controls.
  - The previous and next button icons can be replaced with slot selectors:
    - `previous-slide-icon` and `next-slide-icon`
  - The entire template can be substituted as long as it provides the same elements.
  - Accepts attributes to alter behavior:
    - timeout: time in seconds before moving to next slide. If '0' or not present the slides won't progress automatically.
    - animation: the name of the transition animation to apply. Following animations are provided:
      - none - no animation
      - fade - fades out current slide, then fades in next
      - crossfade - fades out current slide and fades in next at the same time
      - scroll - scrolls out current slide and scrolls in next
      - Custom animations can be added and referenced by class name
    - slide: The current slide being shown.
    - Altering any attribute will reflect in the component state.
    - Raises a custom event: `slide-changed` when the current slide changes.
      - Currently used to show the slide title outside the slide-show element.
 
I've ended up with 80+ commits on this branch, so there may be more I've forgotten 0.o.

Issues:
- Weird bug where the timer occasionally doesn't reset when the slide changes.

Tested in:
- Desktop (Linux and Mac OS):
  - Firefox, Safari, Vivaldi and Chrome.
- Mobile (Android):
  - Chrome, Firefox.

Suggested enhancements:
- Could do with some attention to accessibility. Not my strong point and I would be interested in seeing what someone else can do with it.
- Plenty of scope for improving styling both inside the component and outside it. Anything can be inserted as a slide and since the slideshow functionality is buried in shadow DOM slide styling is largely independent. So go nuts.
- Plenty more features could be added. I'm pretty much spent on this right now, but happy to answer questions and help if anyone else wants to do some enhancements.

## Hacktoberfest 2022:
- [x] Does this PR follow the `intermediate.md` guide and submitted as part of Hacktoberfest 2022?
- [x] Convert slideshow to web component
- [x] Use progressive enhancement to keep functionality if component fails
- [x] Fix transitions issue - (Should be a thing of the past)
- [x] Add alternative animations for slide transitions (and some :))

## Breaking changes
None

## Screenshots
Progressive Enhancement:
image
Full component:
image

### CODE REVIEW

File: server.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+python3 -m http.server
```
Comment (auxfuse) [inline server.sh:1]: This is my only thing to comment on 😆 ...

Is this file needed? @seanyoung247

**Review by auxfuse — CHANGES_REQUESTED** [review]
I am genuinely impressed man. Over and over and over again. This is absolutely fantastic, and I'm so incredibly proud to have you on board with the enhancement of this project. 👏 

Just one thing to query!

Also, nice tough of keeping .txt file for the iframe sliding. I like that idea, and will implement that idea for something that's a bit progressive in work! So thanks 😅

File: server.sh (hunk 1 above)
Comment (seanyoung247) [reply server.sh:1]: Oh, no. I added it because it's a pain to keep typing python3 -m http.server when testing locally. I'll remove.

**Review by auxfuse — APPROVED** [review]
LGTM 👏🎉👏🎉👏🎉
